servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -464,50 +369,27 @@ public void completeFile(long fileId) throws BlockInfoException, FileDoesNotExis
   }
 
   void completeFileInternal(List<Long> blockIds, long fileId, long fileLength, boolean replayed,
-      long opTimeMs) throws FileDoesNotExistException, InvalidPathException {
+      long opTimeMs) throws BlockInfoException, FileDoesNotExistException, InvalidPathException,
+      SuspectedFileSizeException {
     // This function should only be called from within synchronized (mInodeTree) blocks.
     mDependencyMap.addFileCheckpoint(fileId);
     InodeFile inodeFile = (InodeFile) mInodeTree.getInodeById(fileId);
     inodeFile.setBlockIds(blockIds);
-    inodeFile.setCompleted(fileLength);
-    inodeFile.setLastModificationTimeMs(opTimeMs);
-    // Mark all parent directories that have a UFS counterpart as persisted.
-    propagatePersisted(inodeFile, replayed);
-  }
-
-  /**
-   * Propagates the persisted status to all parents of the given inode in the same mount partition.
-   *
-   * @param inode the inode to start the propagation at
-   * @param replayed whether the invocation is a result of replaying the journal
-   * @throws FileDoesNotExistException if a non-existent file is encountered
-   * @throws InvalidPathException if an invalid path is encountered
-   */
-  private void propagatePersisted(Inode inode, boolean replayed) throws FileDoesNotExistException,
-      InvalidPathException {
-    if (!inode.isPersisted()) {
-      return;
-    }
-    Inode handle = inode;
-    while (handle.getParentId() != InodeTree.NO_PARENT) {
-      handle = mInodeTree.getInodeById(handle.getParentId());
-      TachyonURI path = mInodeTree.getPath(handle);
-      if (mMountTable.isMountPoint(path)) {
-        // Stop propagating the persisted status at mount points.
-        break;
-      }
-      if (handle.isPersisted()) {
-        // Stop if a persisted directory is encountered.
-        break;
-      }
-      handle.setPersisted(true);
-      if (!replayed) {
-        writeJournalEntry(new PersistDirectoryEntry(inode.getId(), inode.isPersisted()));
-      }
+    inodeFile.complete(fileLength);
+
+    // TODO(jiri): Do we need this?
+    long currLength = fileLength;
+    for (long blockId : inodeFile.getBlockIds()) {
+      long blockSize = Math.min(currLength, inodeFile.getBlockSizeBytes());
+      mBlockMaster.commitBlockInUFS(blockId, blockSize);
```

Comment:
Is this the right place for this logic? If the file was written in Tachyon only, I don't think we want to call `commitBlockInUFS`?

In general, what should the contract be when `completeFile` is called? Should the block level metadata always be available in the inode (which is the case for Tachyon writes)?